File: SumCalculator/app/src/main/java/com/example/meincrab/sumcalculator/MainActivity.java
package com.example.meincrab.sumcalculator;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import org.w3c.dom.Text;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onClickCalc(View view) {
        Toast.makeText(getApplicationContext(), "Hello World", Toast.LENGTH_SHORT).show();
        EditText num1 = (EditText)findViewById(R.id.number1);
        EditText num2 = (EditText)findViewById(R.id.number2);
        TextView calcText = (TextView)findViewById(R.id.output);

        float number1 = Float.parseFloat(num1.getText().toString());
        float number2 = Float.parseFloat(num2.getText().toString());
        float result = number1 + number2;
        calcText.setText(Float.toString(result));
        Toast.makeText(getApplicationContext(), Float.toString(result), Toast.LENGTH_SHORT).show();
    }
}

File: VideoPlayer/app/src/main/java/com/example/k8684/videoplayer/MainActivity.kt
package com.example.k8684.videoplayer

import android.content.Context
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.android.youtube.player.YouTubeBaseActivity
import com.google.android.youtube.player.YouTubeInitializationResult
import kotlinx.android.synthetic.main.activity_main.*
import com.google.android.youtube.player.YouTubePlayer

class MainActivity : YouTubeBaseActivity() {

    lateinit var youtubePlayerInit: YouTubePlayer.OnInitializedListener

    lateinit var textFromField : EditText
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val youtubeApiKey = "There was a key one Day, but not anymore."
        val playButton = findViewById<Button>(R.id.playButton)
        val getLinkButton = findViewById<Button>(R.id.getLinkButton)

        getLinkButton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                textFromField = findViewById<EditText>(R.id.addressInput)
                //Toast.makeText(getApplicationContext(), textFromField.text, Toast.LENGTH_LONG).show()
                var videoCode = textFromField.getText().toString()
                if (videoCode.length < 30) {
                    videoCode = videoCode.substring(17)
                }
                else {
                    videoCode = videoCode.substring(32)
                }

                Toast.makeText(getApplicationContext(), videoCode, Toast.LENGTH_LONG).show()

                activatePlayer(videoCode)
            }
        })

        playButton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                youtubePlayer.initialize(youtubeApiKey, youtubePlayerInit)
            }
        })
    }
    fun activatePlayer(videoCode : String) {

        youtubePlayerInit = object: YouTubePlayer.OnInitializedListener{
            override fun onInitializationSuccess(p0: YouTubePlayer.Provider?, youtubePlayer: YouTubePlayer?, p2: Boolean) {
                youtubePlayer?.loadVideo(videoCode)
            }

            override fun onInitializationFailure(p0: YouTubePlayer.Provider?, p1: YouTubeInitializationResult?) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }
        }




    }
}